src/Workleap.Extensions.Mongo/MongoServiceCollectionExtensions.cs

```diff
@@ -37,6 +43,86 @@ public static MongoBuilder AddMongo(this IServiceCollection services, Action<Mon
         return new MongoBuilder(services);
     }
 
+    /// <summary>
+    /// Adds Configuration based support for Mongo Collection Names, Index Providers and Class maps.
+    /// Each Collection requires a name. Index Providers are optional, as are Class maps.
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <param name="assemblies">Assemblies in which we can find the implementations of IMongoCollectionConfiguration</param>
+    /// <returns></returns>
+    /// <exception cref="InvalidOperationException"></exception>
+    /// <exception cref="ArgumentNullException"></exception>
+    public static MongoBuilder AddCollectionConfigurations(this MongoBuilder builder, params Assembly[] assemblies)
+    {
+        var configurationTypes = GetMongoCollectionConfigurationTypes(assemblies);
+
+        foreach (var configurationType in configurationTypes)
+        {
+            var (concreteType, configurationInterface) = configurationType;
+
+            if (RegisteredConfigurations.Contains(concreteType))
+            {
+                continue;
+            }
+
+            var configuration = GetMongoCollectionConfiguration(concreteType);
+
+            var documentType = configurationInterface.GetGenericArguments().Single();
+            var builderType = typeof(MongoCollectionBuilder<>).MakeGenericType(documentType);
+
+            if (Activator.CreateInstance(builderType) is not MongoCollectionBuilder configurationBuilder)
+            {
+                throw new InvalidOperationException($"Cannot create {builderType}");
+            }
+            
+            var metadata = GetMongoCollectionMetadata(documentType, configuration, configurationBuilder);
+
+            if (string.IsNullOrWhiteSpace(metadata.CollectionName))
+            {
+                throw new ArgumentNullException($"{builderType} must specify a CollectionName");
+            }
+            
+            MongoCollectionNameCache.SetCollectionName(documentType, metadata.CollectionName!);
```

Comment:
Should we just set metadata.CollectionName as non nullable? Otherwise we open the door to setting to null and could the feature

Reply:
Cant be not nullable since it has to be null before the user calls .CollectionName()